elasticdl/python/common/args.py

```diff
@@ -383,3 +329,146 @@ def _get_attr(instance, attribute):
+    )
+    parser.add_argument(
+        "--loss",
+        type=str,
+        default="loss",
+        help="The name of the loss function defined in the model file",
+    )
+    parser.add_argument(
+        "--optimizer",
+        type=str,
+        default="optimizer",
+        help="The name of the optimizer defined in the model file",
+    )
+    parser.add_argument(
+        "--eval_metrics_fn",
+        type=str,
+        default="eval_metrics_fn",
+        help="The name of the evaluation metrics function defined "
+        "in the model file",
+    )
+    parser.add_argument(
+        "--model_def",
+        type=str,
+        required=True,
+        help="The import path to the model definition function/class in the "
+        'model zoo, e.g. "cifar10_subclass.cifar10_subclass.CustomModel"',
+    )
+    parser.add_argument(
+        "--model_params",
+        type=str,
+        default="",
+        help="The dictionary of model parameters in a string that will be "
+        'used to instantiate the model, e.g. "param1=1,param2=2"',
+    )
+    parser.add_argument(
+        "--get_model_steps",
+        type=int,
+        default=1,
+        help="Worker will get_model from PS every these steps.",
+    )
+
+
+def parse_master_args(master_args=None):
+    parser = argparse.ArgumentParser(description="ElasticDL Master")
+    parser.add_argument(
+        "--port",
+        default=50001,
+        type=pos_int,
+        help="The listening port of master",
+    )
+    parser.add_argument(
+        "--worker_image", help="Docker image for workers", default=None
+    )
+    parser.add_argument(
+        "--worker_pod_priority", help="Priority requested by workers"
+    )
+    parser.add_argument(
+        "--prediction_data_dir",
+        help="Prediction data directory. Files should be in RecordIO format",
+        default="",
+    )
+    add_common_params(parser)
+    add_train_params(parser)
+
+    args, _ = parser.parse_known_args(args=master_args)
+    print_args(args, groups=ALL_ARGS_GROUPS)
+    logger.warning("Unknown arguments: %s", _)
+
+    if all(
+        v == "" or v is None
+        for v in [
+            args.training_data_dir,
+            args.evaluation_data_dir,
+            args.prediction_data_dir,
+        ]
+    ):
+        raise ValueError(
+            "At least one of the data directories needs to be provided"
+        )
+
+    if args.prediction_data_dir and (
+        args.training_data_dir or args.evaluation_data_dir
+    ):
+        raise ValueError(
+            "Running prediction together with training or evaluation "
+            "is not supported"
+        )
+    if args.prediction_data_dir and not args.checkpoint_filename_for_init:
+        raise ValueError(
+            "checkpoint_filename_for_init is required for running "
+            "prediction job"
+        )
+
+    return args
+
+
+def parse_worker_args(worker_args=None):
```

Comment:
I am not sure if it makes sense to put this here since it's common/args.py and these are worker specific.

Reply:
You can assume that `common/args.py` is used to manage arguments of ElasticDL. It seems better to place similar functions in the same place to avoid duplication.

Reply:
Okay